Remotion/Data/DomainObjects.UnitTests/DomainObjectExtensionsTest.cs

```diff
@@ -81,15 +81,14 @@ public void GetDefaultTransactionContext ()
     {
       IDomainObject domainObject = DomainObjectMother.CreateFakeObject<Order>();
 
-      var actualTransactionContext = (DomainObjectTransactionContext)domainObject.GetDefaultTransactionContext();
+      var actualTransactionContext = domainObject.GetDefaultTransactionContext();
 
       var expectedTransactionContext =
-          (DomainObjectTransactionContext)domainObject.TransactionContext[domainObject.RootTransaction.ActiveTransaction];
+          domainObject.TransactionContext[domainObject.RootTransaction.ActiveTransaction];
 
       Assert.That(actualTransactionContext.ClientTransaction, Is.EqualTo(expectedTransactionContext.ClientTransaction));
       Assert.That(actualTransactionContext.DomainObject, Is.EqualTo(expectedTransactionContext.DomainObject));
-      Assert.That(actualTransactionContext, Is.Not.EqualTo(expectedTransactionContext), "DomainObjectTransactionContext cannot be compared by reference at the moment.");
-
+      Assert.That(actualTransactionContext, Is.EqualTo(expectedTransactionContext));
```

Comment:
Since the `transactionContext` was a reference type previously, the equals implementation would not return true. The new implementation uses structs, which by default implement value equality -> we actually only need the last Assertion, but for readability I would probably keep the other 2 as well.